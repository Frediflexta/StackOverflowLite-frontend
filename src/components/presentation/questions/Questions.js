import React from 'react';
import { Link } from 'react-router-dom';
import propTypes from 'prop-types';
import './Questions.scss';

const Questions = (props) => {
  const {
    question
  } = props;

  return (
    <div className="content-container">
      <div className="votes">
        {/* <div className="v-counts">
          <span>
            {question.voteCount}
          </span>
        </div>
        <div className="quote">vote(s)</div> */}
      </div>
      <div className="answers">
        <div className="ans-counts">
          <span>
            {question.totalanswers}
          </span>
        </div>
        <div className="quote">answers</div>
      </div>
      <div className="ques-details">
        <h3><Link to={`/questions/${question.id}`}>
          {question.questitle}
        </Link></h3>
        <p>{question.quesbody}</p>
      </div>
    </div>
  );
};

Questions.propTypes = {
  question: propTypes.object,
};


export default Questions;
