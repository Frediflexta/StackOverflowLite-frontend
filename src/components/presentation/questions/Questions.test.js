import React from 'react';
// import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import Questions from './Questions';

describe('<Questions /> shallow rendering tests', () => {
  const props = {
    question: {
      totalanswers: '1',
      id: '2',
      questitle: 'something',
      quesbody: 'something here too'
    }
  };
  const wrapper = shallow(
    <Questions {...props} />
  );

  it('should render Questions with props successfully', () => {
    expect(wrapper.find('.content-container').exists()).toBe(true);
    expect(wrapper.find('.quote').exists()).toBe(true);
  });

  it('matches the snapshot', () => {
    const tree = shallow(<Questions {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
