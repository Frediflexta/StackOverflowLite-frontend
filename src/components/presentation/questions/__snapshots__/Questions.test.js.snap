// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Questions /> shallow rendering tests matches the snapshot 1`] = `
<div
  className="content-container"
>
  <div
    className="votes"
  />
  <div
    className="answers"
  >
    <div
      className="ans-counts"
    >
      <span>
        1
      </span>
    </div>
    <div
      className="quote"
    >
      answers
    </div>
  </div>
  <div
    className="ques-details"
  >
    <h3>
      <Link
        replace={false}
        to="/questions/2"
      >
        something
      </Link>
    </h3>
    <p>
      something here too
    </p>
  </div>
</div>
`;
